### PR TITLE
Updates mobile SM API docs with additional details

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description of change
Adds more details to several of the SM API docs. 

No testing or other code concerns, exclusively a documentation update

### CODE REVIEW

File: modules/mobile/docs/schemas/SecureMessageReplyRequest.yml [hunk 1]
```diff
@@ -1,6 +1,22 @@
 type: object
 required:
+  - recipient_id 
+  - category
+  - subject
   - body
 properties:
+  recipient_id:
```
Comment (patrickvinograd) [inline modules/mobile/docs/schemas/SecureMessageReplyRequest.yml:8]: Actually I was wondering if recipient_id is required for replies, or if there's validation that the recipient matches the sender of the replied-to message. That would be good to clarify.

**Comment by patrickvinograd** [comment]
Happy to have additional doc updates incorporated into this PR.

File: modules/mobile/docs/schemas/SecureMessageReplyRequest.yml (hunk 1 above)
Comment (narin) [reply modules/mobile/docs/schemas/SecureMessageReplyRequest.yml:8]: In my testing, I've found that aside from file uploads, only `body` is required/allowed for replies.  `recipient_id` is ignored on the MHV side.  I'll make an update to our API's validation.

**Review by narin — APPROVED** [review]
Fixed a couple merge conflicts. LGTM!

**Pull request merged**